src/net_processing.cpp

```diff
@@ -5194,6 +5197,62 @@ void PeerManagerImpl::CheckForStaleTipAndEvictPeers()
     }
 }
 
+void PeerManagerImpl::CheckForLimitedMempoolAndEvictPeers()
+{
+    LOCK(cs_main);
+    // It only makes sense once we're out of IBD, because otherwise our mempool
+    // is not a good source for fee estimation anyway.
+    // TODO: assumeUTXO?
+    if (!CanDirectFetch()) return;
+
+    // Don't bother if we are still seeking for outbound peers: eviction
+    // instead will reduce our chances to get blocks and transactions from
+    // them.
+    // TODO this could be more nuanced.
+    if (m_connman.GetExtraFullOutboundCount() < 0) return;
+
+    // Ideally we want to compare mempool sizes, not fee filters.
+    // Otherwise we easily get confused: e.g. at empty mempools this is less
+    // critical, but that'd be impossible to account for.
+    const auto our_min_feerate = WITH_LOCK(m_mempool.cs, return m_mempool.GetMinFee());
+
+    std::vector<std::pair<CNode*, CAmount>> peer_fee_filters;
+
+    size_t around_our_percent_fee_filter = 0;
+
+    m_connman.ForEachNode([&](CNode* pnode) {
+        if (!pnode->IsFullOutboundConn() || pnode->fDisconnect) return;
+        auto peer = GetPeerRef(pnode->GetId());
+
+        if (auto tx_relay = peer->GetTxRelay(); tx_relay != nullptr) {
+            if (tx_relay->m_fee_filter_received) {
+                peer_fee_filters.push_back(std::make_pair(pnode, tx_relay->m_fee_filter_received.load()));
```

Comment:
This looks like a boolean and got me confused "if fee filter has been received". It is actually an integer. When dealing with atomics it is better to read once into a local variable and later refer the local variable to avoid the value changing in the meantime.

```suggestion
            const auto minfee = tx_relay->m_fee_filter_received.load();
            if (minfee > 0) {
                peer_fee_filters.push_back(std::make_pair(pnode, minfee));
```